### PR TITLE
Make OAuth2 config client-secret sensitive

### LINKED ISSUE #6889: [oauth2] config option client-secret should be sensitive
## Problem

When configuring oauth2 auth:

```
http-server.authentication.oauth2.client-secret=xxx-yyy-zzz
```

Then starts the coordinator node, it dumps the above sensitive info into the logs like:

```
2021-02-14T01:30:27.415Z	INFO	main	Bootstrap	http-server.authentication.oauth2.client-secret                             ----                                                                       xxx-yyy-zzz                                                 Client secret
```

## Expected result

When starting coordinator node with oauth2, it should hide the value of `client-secret` but show `REDACTED` like:

```
2021-02-14T01:30:27.415Z	INFO	main	Bootstrap	http-server.authentication.oauth2.client-secret                             ----                                                                       [REDACTED]                                                 Client secret

```

## Proposed fix

Add `@ConfigSecuritySensitive` here:

https://github.com/trinodb/trino/blob/7b83702fcc1c04d49abee6a60f784653979f6325/core/trino-main/src/main/java/io/trino/server/security/oauth2/OAuth2Config.java#L124-L126

### PR DESCRIPTION
**Pull request by shawnzhu**
Fixes #6889

### CODE REVIEW

File: core/trino-main/src/test/java/io/trino/server/security/oauth2/TestOAuth2Config.java [hunk 1]
```diff
@@ -14,20 +14,33 @@
...
 import java.util.Map;
 
+import static com.google.common.collect.ImmutableList.toImmutableList;
 import static io.airlift.configuration.testing.ConfigAssertions.assertFullMapping;
 import static io.airlift.configuration.testing.ConfigAssertions.assertRecordedDefaults;
 import static io.airlift.configuration.testing.ConfigAssertions.recordDefaults;
 
 public class TestOAuth2Config
 {
+    private static List<String> findSensitiveConfigName(Class configClass)
```
Comment (kokosing) [inline core/trino-main/src/test/java/io/trino/server/security/oauth2/TestOAuth2Config.java:36]: Please move this method to the bottom, below the usage.

**Review by kokosing — APPROVED** [review]
% comment

File: core/trino-main/src/test/java/io/trino/server/security/oauth2/TestOAuth2Config.java (hunk 1 above)
Comment (findepi) [reply core/trino-main/src/test/java/io/trino/server/security/oauth2/TestOAuth2Config.java]: I don't think there should be a test for this. Its just testing annotation existence.

**Comment by findepi** [comment]
FYI, I merged the annotation and closed the #6889 so that we can continue discussion here with less urgency.

**Review by lukasz-walkiewicz — APPROVED** [review]

File: core/trino-main/src/test/java/io/trino/server/security/oauth2/TestOAuth2Config.java (hunk 1 above)
Comment (kokosing) [reply core/trino-main/src/test/java/io/trino/server/security/oauth2/TestOAuth2Config.java]: I was thinking the same way. But one may say that entire `TestOAuth2Config` goes into the same bucket. I am fine either way.

Comment (findepi) [reply core/trino-main/src/test/java/io/trino/server/security/oauth2/TestOAuth2Config.java]: It's an existing convention. We can consider changing the convention either way -- like removing those `Test*Config` classes, or adding assertions for `@ConfigSecuritySensitive` existence. 
If we want to do so, we should have some sort of high-level agreement.

**Comment by shawnzhu** [comment]
> FYI, I merged the annotation and closed the #6889 so that we can continue discussion here with less urgency.

@findepi thanks! let me rebase

File: core/trino-main/src/test/java/io/trino/server/security/oauth2/TestOAuth2Config.java (hunk 1 above)
Comment (kokosing) [reply core/trino-main/src/test/java/io/trino/server/security/oauth2/TestOAuth2Config.java]: @shawnzhu Is this ok to you if just drop this test?

Comment (shawnzhu) [reply core/trino-main/src/test/java/io/trino/server/security/oauth2/TestOAuth2Config.java]: @kokosing sure! It's just a test to make sure `ConfigSecuritySensitive` annotation is added to the desired method, as a requirement.

I'd hope to move the method `findSensitiveConfigName()` to somewhere like `trino-testing` or even upperstream airlift (don't know where it is)

**Comment by shawnzhu** [comment]
close to follow up in #6906

**Pull request closed without merging**